apollo-router/src/plugins/telemetry/otlp.rs

```diff
@@ -189,39 +159,25 @@ fn header_map(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Sch
 }
 
 impl GrpcExporter {
-    // Return a TlsConfig if it has something actually set.
-    pub(crate) fn try_from(
-        self,
-        endpoint: &Option<Uri>,
-    ) -> Result<Option<ClientTlsConfig>, BoxError> {
-        if let Some(endpoint) = endpoint {
-            let endpoint = endpoint.to_string().parse::<Url>().map_err(|e| {
-                BoxError::from(format!("invalid GRPC endpoint {}, {}", endpoint, e))
-            })?;
-            let domain_name = self.default_tls_domain(&endpoint);
-
-            if self.ca.is_some()
-                || self.key.is_some()
-                || self.cert.is_some()
-                || domain_name.is_some()
```

Comment:
If we check that they are Some then it doesn't make sense to use `with` and `try_with`